packages/forms/src/model.ts

```diff
@@ -444,9 +454,92 @@ export abstract class AbstractControl {
    * `updateValueAndValidity()` for the new validation to take effect.
    *
    */
-  setAsyncValidators(newValidator: AsyncValidatorFn|AsyncValidatorFn[]|null): void {
-    this._rawAsyncValidators = newValidator;
-    this._composedAsyncValidatorFn = coerceToAsyncValidator(newValidator);
+  setAsyncValidators(validators: AsyncValidatorFn|AsyncValidatorFn[]|null): void {
+    this._rawAsyncValidators = validators;
+    this._composedAsyncValidatorFn = coerceToAsyncValidator(validators);
+  }
+
+  /**
+   * Add a sync validator or validators to this control, without affecting other validators.
+   *
+   * When you add or remove a validator at run time, you must call
+   * `updateValueAndValidity()` for the new validation to take effect.
+   *
+   * @param validators The new validator function or functions to add to this control.
+   */
+  addValidators(validators: ValidatorFn|ValidatorFn[]): void {
+    const newValidators = makeValidatorsArray(validators).filter(v => !this.hasValidator(v));
+    const oldValidators = makeValidatorsArray(this._rawValidators);
+    this.setValidators(oldValidators.concat(newValidators));
+  }
+
+  /**
+   * Add an asynchronous validator or validators to this control, without affecting other
+   * validators.
+   *
+   * When you add or remove a validator at run time, you must call
+   * `updateValueAndValidity()` for the new validation to take effect.
+   *
+   * @param validators The new async validator function or functions to add to this control.
+   */
+  addAsyncValidators(validators: AsyncValidatorFn|AsyncValidatorFn[]): void {
+    const newValidators = makeValidatorsArray(validators).filter(v => !this.hasAsyncValidator(v));
+    const oldValidators = makeValidatorsArray(this._rawAsyncValidators);
+    this.setAsyncValidators(oldValidators.concat(newValidators));
+  }
+
+  /**
+   * Remove a sync validator from this control, without affecting other validators. Validators are
+   * compared by function reference; you must pass a reference to the exact same validator
+   * function as the one that was originally set.
+   *
+   * When you add or remove a validator at run time, you must call
+   * `updateValueAndValidity()` for the new validation to take effect.
+   *
+   * @param validators The validator or validators to remove.
+   */
+  removeValidators(validators: ValidatorFn|ValidatorFn[]): void {
+    const newValidators = makeValidatorsArray(validators);
```

Comment:
I think we should rename the const:
```suggestion
    const validatorsToRemove = makeValidatorsArray(validators);
```